src/NHibernate/Util/ServiceProviderExtensions.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NHibernate.Util
+{
+	internal static class ServiceProviderExtensions
+	{
+		public static object GetInstance(this IServiceProvider serviceProvider, System.Type serviceType)
+		{
+			// 6.0 TODO throw a meaningful exception instead of using the Activator
+			return serviceProvider.GetService(serviceType) ?? Activator.CreateInstance(serviceType);
```

Comment:
I did this because the `Activator.CreateInstance` throws a `MissingMethodException` for an abstract or interface type where now a `HibernateServiceProviderException` is thrown instead. As this can only happen when the user incorrectly configures a service, we shouldn't bother and I agree with you that we don't need to wait for 6.0.